### PR TITLE
Replace lambda statements with expression lambdas where applicable

### PR DESCRIPTION
**Pull request by Shadow-Devil**
## Purpose
Remove useless curly braces around statement and then remove useless return keyword.
The right-hand side of a lambda expression can be written in two ways:

1. Expression notation: the right-hand side is as an expression, such as in (a, b) → a + b
2. Block notation: the right-hand side is a conventional function body with a code block and an optional return statement, such as in (a, b) → {return a + b;}
By convention, expression notation is preferred over block notation. Block notation must be used when the function implementation requires more than one statement. However, when the code block consists of only one statement (which may or may not be a return statement), it can be rewritten using expression notation.

This convention exists because expression notation has a cleaner, more concise, functional programming style and is regarded as more readable.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by hasithaa — APPROVED** [review]
LGTM

**Comment by hasithaa** [comment]
@Shadow-Devil there is a conflict. will you be able to fix it?

File: compiler/ballerina-lang/src/main/java/org/ballerinalang/toml/model/Manifest.java [hunk 1]
```diff
@@ -105,10 +105,10 @@ public String getTargetPlatform(String moduleName) {
...
             }
             // Check if module have platform specific libraries
-            List<Library> deps = platform.libraries.stream().filter(library -> {
-                return library.getModules() == null ||
-                        Arrays.stream(library.getModules()).anyMatch(moduleName::equals);
-            }).collect(Collectors.toList());
+            List<Library> deps = platform.libraries.stream().filter(library ->
+                library.getModules() == null ||
+                        Arrays.stream(library.getModules()).anyMatch(moduleName::equals)
+            ).collect(Collectors.toList());
```
Comment (chiranSachintha) [inline compiler/ballerina-lang/src/main/java/org/ballerinalang/toml/model/Manifest.java:111]:
```
List<Library> deps = platform.libraries.stream().filter(library -> library.getModules() == null ||
                                                    Arrays.asList(library.getModules()).contains(moduleName)).toList();
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java [hunk 2]
```diff
@@ -2947,10 +2946,9 @@ public BLangNode transform(DoStatementNode doStatementNode) {
         BLangBlockStmt bLBlockStmt = (BLangBlockStmt) doStatementNode.blockStatement().apply(this);
         bLBlockStmt.pos = getPosition(doStatementNode.blockStatement());
         bLDo.setBody(bLBlockStmt);
-        doStatementNode.onFailClause().ifPresent(onFailClauseNode -> {
-            bLDo.setOnFailClause(
-                    (org.ballerinalang.model.clauses.OnFailClauseNode) (onFailClauseNode.apply(this)));
-        });
+        doStatementNode.onFailClause().ifPresent(onFailClauseNode -> bLDo.setOnFailClause(
+                (org.ballerinalang.model.clauses.OnFailClauseNode) (onFailClauseNode.apply(this)))
+        );
```
Comment (chiranSachintha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java:2951]:
```suggestion
        doStatementNode.onFailClause().ifPresent(onFailClauseNode -> bLDo.setOnFailClause(
                (org.ballerinalang.model.clauses.OnFailClauseNode) (onFailClauseNode.apply(this))));
```

Comment (chiranSachintha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java:2951]: Please fix in other places as well.

**Review by chiranSachintha — APPROVED** [review]

**Pull request merged**